### DISCUSSION TITLE
Welcome to Zod discussions!

### DISCUSSION THREAD

**Discussion by colinhacks** [colinhacks/zod#263] [General]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by timsgardner** [comment]
Hello, I'm trying Zod out and really enjoying it so far. The README docs for `parse` state:

> Given any Zod schema, you can call its .parse method to check data is valid. If it is, a value is returned with full type information! Otherwise, an error is thrown.

What does "with full type information" mean here? I couldn't find examples in the README of anything being returned from `parse` beyond the input value (or a clone of it). Thanks!

**Comment by colinhacks** [comment]
It just means the returned value is statically typed. So you can, for instance, hover over it and see it's type. You'll also get autocompletion, etc if you're using an IDE that supports it. So Zod is great for parsing, say, payloads to an API endpoint in order to guarantee certain properties are present:

const schema = z.object({ name: z.string() });

const rawPayload: any = { /* some blob of json */ };
const parsedPayload = schema.parse(rawPayload);

parsedPayload.name; // => string

**Reply by timsgardner** [reply]
I see, that makes sense, thanks.

**Comment by bingtimren** [comment]
Hi, thanks for making Zod!

Both joi and yup has an option "abortEarly", which instructs the validator to stop validation upon encountering the first issue. Is there a similar option in Zod? Maybe the "check" method do exactly this? If this is true, it worth mentioning this in README. Thanks!

Bing
